RecoEgamma/ElectronIdentification/plugins/ElectronHEEPIDValueMapProducer.cc

```diff
@@ -204,16 +230,23 @@ int ElectronHEEPIDValueMapProducer::nrSaturatedCrysIn5x5(const reco::GsfElectron
 
 }
 
-float ElectronHEEPIDValueMapProducer::calTrkIso(const reco::GsfElectron& ele,		  
-						const edm::View<reco::GsfElectron>& eles,
-						const std::vector<edm::Handle<pat::PackedCandidateCollection> >& handles)
+float ElectronHEEPIDValueMapProducer::
+calTrkIso(const reco::GsfElectron& ele,		  
+	  const edm::View<reco::GsfElectron>& eles,
+	  const std::vector<edm::Handle<pat::PackedCandidateCollection> >& handles,
+	  const std::vector<EleTkIsolFromCands::PIDVeto>& pidVetos)const
 {
   if(ele.gsfTrack().isNull()) return std::numeric_limits<float>::max();
   else{
     float trkIso=0.; 
-    for(auto& handle: handles){
+    for(size_t handleNr=0;handleNr<handles.size();handleNr++){
+      auto& handle = handles[handleNr];
       if(handle.isValid()){
-	trkIso+= trkIsoCalc_.calIsolPt(*ele.gsfTrack(),*handle,eles);
+	if(handleNr<pidVetos.size()){
+	  trkIso+= trkIsoCalc_.calIsolPt(*ele.gsfTrack(),*handle,pidVetos[handleNr]);
+	}else{
```

Comment:
shouldn't this handle-pidveto size mismatch be detected in the constructor (instead of being checked every electron) ?

Reply:
So yes you are correct. And it is there.  However to due how it auto detects miniAOD or AOD and miniAOD and AOD can have different numbers of input collections, it means the loop over the handle numbers is not completely trivial.  If I made a mistake in coding, there was a risk of an invalid memory access due to this so I put an additional check there. 

Now that I've run it and there is no bug in the code, I can just remove it if you like as its not doing anything and will always evaluate to true. However I find it nice that a new reader of the code can quickly see for sure that this is not going out of bounds rather than working through the logic. However it does come with a computing cost, however small and so I'll do what you wish me to do here.

Reply:
can the auto-detection be protected at compile time to always give a value?
Otherwise, it seems that every time a loop is needed, there should be a size check.

Reply:
so really the only thing this is doing is telegraphing my intention and making sure future code readers can quickly see that its okay and is not accessing out of bounds.  The issue comes from the fact that the number of handles can be bigger than the number of vetos but the number of valid handles will always be the same. 

It has no other use apart from guarding against bugs.  So I can remove it if you wish.

Reply:
fine to stay, no strong preference